### PR TITLE
DATAGO-79772: Cloud managed EMA must not stream configPush logs to EP Core

### PR DESCRIPTION
**Pull request by rudraneel-chakraborty**
### What is the purpose of this change?

Currently, Self managed EMA streams Terraform execution logs to ep-core. These logs end up in Datadog and aids in debuggability. Cloud managed EMA already has DD integration. So we intend to disable the log streaming feature if the ema is running in cloud managed mode

### How was this change implemented?
Java code change

### How was this change tested?

IT

### Is there anything the reviewers should focus on/be aware of?

    ...

### CODE REVIEW

File: service/application/src/main/java/com/solace/maas/ep/event/management/agent/processor/CommandLogStreamingProcessor.java [hunk 1]
```diff
@@ -50,28 +49,6 @@ public CommandLogStreamingProcessor(CommandLogsPublisher commandLogsPublisher,
...
-            if (Files.exists(path)) {
-                Files.delete(path);
-            }
-        } catch (IOException e) {
-            log.warn("Error while deleting execution log at {}", path, e);
-            return false;
-        }
-        return true;
-    }
-
```
Comment (rudraneel-chakraborty) [inline service/application/src/main/java/com/solace/maas/ep/event/management/agent/processor/CommandLogStreamingProcessor.java:73]: Extracted these methods as is to `CommandManager` as these methods are better suited there

File: service/application/src/test/java/com/solace/maas/ep/event/management/agent/TestConfig.java [hunk 2]
```diff
@@ -161,29 +160,14 @@ public CommandLogsPublisher getComaCommandLogsPublisher() {
...
-                                                                         ObjectMapper objectMapper) {
-        return new CommandLogStreamingProcessor(commandLogsPublisher, eventPortalProperties, objectMapper);
-    }
-
-    @Bean
-    @Qualifier("mockedCommandLogStreamingProcessor")
-    @Primary
-    public CommandLogStreamingProcessor mockedCommandLogStreamingProcessor() {
-        return mock(CommandLogStreamingProcessor.class);
-    }
```
Comment (rudraneel-chakraborty) [inline service/application/src/test/java/com/solace/maas/ep/event/management/agent/TestConfig.java:177]: Not needed

File: service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java [hunk 3]
```diff
@@ -45,14 +48,14 @@ public class CommandManager {
     private final MessagingServiceDelegateService messagingServiceDelegateService;
     private final EventPortalProperties eventPortalProperties;
     private final ThreadPoolTaskExecutor configPushPool;
-    private final CommandLogStreamingProcessor commandLogStreamingProcessor;
+    private final Optional<CommandLogStreamingProcessor> commandLogStreamingProcessorOpt;
```
Comment (rudraneel-chakraborty) [inline service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java:51]: commandLogStreamingProcessor is absent for cloud managed mode. So making it optional

File: service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java [hunk 4]
```diff
@@ -106,7 +109,10 @@ public void configPush(CommandRequest request) {
                 for (Command command : bundle.getCommands()) {
                     Path executionLog = executeCommand(request, command, envVars);
                     if (executionLog != null) {
-                        streamCommandExecutionLogToEpCore(request, command, executionLog);
+                        if (commandLogStreamingProcessorOpt.isPresent()) {
+                            streamCommandExecutionLogToEpCore(request, command, executionLog);
+                        }
```
Comment (rudraneel-chakraborty) [inline service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java:114]: Only stream if the commandLogStreamingProcessorOpt is present (e.g. in self managed mode)

File: service/application/src/test/java/com/solace/maas/ep/event/management/agent/commandManager/CloudManagedEMACommandManagerTests.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, properties = {
+        "eventPortal.gateway.messaging.standalone=false",
+        "eventPortal.managed=true",
+        "eventPortal.incomingRequestQueueName = ep_core_ema_requests_123456_123123",
+        "event-portal.gateway.messaging.rto-session=false"
+})
+@DirtiesContext(classMode = BEFORE_EACH_TEST_METHOD)
+//CPD-OFF
+class CloudManagedEMACommandManagerTests {
+
```
Comment (rudraneel-chakraborty) [inline service/application/src/test/java/com/solace/maas/ep/event/management/agent/commandManager/CloudManagedEMACommandManagerTests.java:51]: This needs to be a separate class as we want to load a new application context in  cloud managed mode

File: service/application/src/main/resources/application.yml [hunk 6]
```diff
@@ -4,393 +4,75 @@ springdoc:
...
-  topicPrefix: ${EP_TOPIC_PREFIX:sc/ep/runtime}
+  organizationId: ${EP_ORGANIZATION_ID:localrcorg}
+  runtimeAgentId: ${EP_RUNTIME_AGENT_ID:3jczd7fk0xv}
+  incomingRequestQueueName: "ep_core_ema_requests_fakedcid"
+  managed: true
   gateway:
-    id: gateway
-    name: evmr1
+    id: f0ydyzsr1jp
+    name: POCCloudManagedEMAEVMR
```
Comment (moodiRealist) [inline service/application/src/main/resources/application.yml:62]: Please undo these changes.

**Review by moodiRealist — COMMENTED** [review]
LGTM, but application.properties file must be rolled back before I can approve.

File: service/application/src/main/resources/application.yml (hunk 6 above)
Comment (rudraneel-chakraborty) [reply service/application/src/main/resources/application.yml]: Done. See slack message how I mitigated the impact

File: service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java [hunk 7]
```diff
@@ -37,6 +38,9 @@ public class TerraformManager {
     private final TerraformProperties terraformProperties;
     private final TerraformClientFactory terraformClientFactory;
 
+
+    private boolean isManagedAgent;
```
Comment (moodiRealist) [inline service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java:42]: Where is this boolean used?

Comment (rudraneel-chakraborty) [reply service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java]: Removed. Good Catch

File: service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java [hunk 8]
```diff
@@ -7,6 +7,7 @@
 import com.solace.maas.ep.event.management.agent.plugin.command.model.CommandResult;
 import com.solace.maas.ep.event.management.agent.plugin.command.model.JobStatus;
 import com.solace.maas.ep.event.management.agent.plugin.constants.RouteConstants;
+import com.solace.maas.ep.event.management.agent.plugin.messagingService.event.EventProperty;
```
Comment (moodiRealist) [inline service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java:10]: I guess this library doesn't have PMD checks. Otherwise, it should have failed the build for un-used imports. 

Please just roll-back this file considering there is no relevant change here.

**Review by moodiRealist — APPROVED** [review]
